compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2769,6 +2805,146 @@ BType getTypeForUnionTypeMembersAssignableToType(BUnionType unionType, BType tar
         }
     }
 
+    BType getTypeIntersectedWithMapType(BMapType mapType, BType targetType, SymbolEnv env) {
+        if (targetType.tag != TypeTags.MAP && targetType.tag != TypeTags.RECORD) {
+            return symTable.semanticError;
+        }
+        if (targetType.tag == TypeTags.MAP) {
+            BType intersectionConstraintType = getTypeIntersection(mapType.constraint,
+                    ((BMapType) targetType).constraint, env);
+            if (intersectionConstraintType == symTable.semanticError) {
+                return symTable.semanticError;
+            }
+            return new BMapType(TypeTags.MAP, intersectionConstraintType, null);
+        }
+        return getTypeIntersectedWithRecordType((BRecordType) targetType, mapType, env);
```

Comment:
Given that it's just two case can't we do
```suggestion
        int targetTypeTag = targetType.tag;
        
        if (targetTypeTag == TypeTags.MAP) {
            BType intersectionConstraintType = getTypeIntersection(mapType.constraint,
                    ((BMapType) targetType).constraint, env);
            if (intersectionConstraintType == symTable.semanticError) {
                return symTable.semanticError;
            }
            return new BMapType(TypeTags.MAP, intersectionConstraintType, null);
        }
        
        if (targetTypeTag == TypeTags.RECORD) {
            return getTypeIntersectedWithRecordType((BRecordType) targetType, mapType, env);
        }
        
        return symTable.semanticError;
```

Reply:
Changed